### PR TITLE
manager: rename dbus method

### PR DESCRIPTION
**Pull request by keszybz**
Fixes #24989.

### CODE REVIEW

**Comment by poettering** [comment]
should also be changed to do the memfd thing. Or to just return info a about single unit

**Comment by keszybz** [comment]
Fine.

File: src/analyze/analyze-dump.c [hunk 1]
```diff
@@ -29,51 +29,114 @@ static int dump_fallback(sd_bus *bus) {
...
         if (r < 0)
                 return bus_log_create_error(r);
 
         r = sd_bus_call(bus, m, 0, &error, &reply);
         if (r < 0)
-                return log_error_errno(r, "Failed to issue method call DumpUnitsMatchingPatterns: %s",
+                return log_error_errno(r, "Failed to call DumpUnitsMatchingPatternsByFileDescriptor: %s",
                                        bus_error_message(&error, r));
 
-        r = sd_bus_message_read(reply, "s", &text);
```
Comment (poettering) [inline src/analyze/analyze-dump.c:64]: maybe move everything in this function from here on to the end into its own helper call `bus_read_fd_and_output()` or so? after all dump() above has the same 9 lines at the end.

File: src/analyze/analyze-dump.c [hunk 2]
```diff
@@ -82,28 +145,23 @@ int verb_dump(int argc, char *argv[], void *userdata) {
...
+                        r = dump_patterns(bus, patterns);
+                else
+                        r = dump(bus);
+        }
+        if (r == 0) {
+                if (patterns)
+                        r = dump_patterns_fallback(bus, patterns);
+                else
+                        r = dump_fallback(bus);
+        }
```
Comment (poettering) [inline src/analyze/analyze-dump.c]:
```c
r = sd_bus_can_send(bus, SD_BUS_TYPE_UNIX_FD);
if (r < 0)
        return log_error_errno(r, "Unable to determine if bus connection supports fd passing: %m");
if (r > 0)
        r = patterns ? dump_patterns(bus, patterns) : dump(bus);
if (r == 0) /* wasn't supported */
        r = patterns ? dump_patterns_fallback(bus, patterns) : dump_fallback(bus);
```

Comment (keszybz) [reply src/analyze/analyze-dump.c]: Ah, nice.

**Comment by fbuihuu** [comment]
It looks good, thanks @keszybz for tackling the remaining points.

**Review by fbuihuu — APPROVED** [review]

**Comment by keszybz** [comment]
Let's merge. This clearly passes compilation tests, and we don't test dump output anywhere, so additional CI isn't very useful.

**Pull request merged**